### DISCUSSION TITLE
Why can't Avalonia bind to `internal` members?

### DISCUSSION THREAD

**Discussion by colejohnson66** [AvaloniaUI/Avalonia#16906] [General]
For our application, we want to use ResX localization in our DLL. Everything works fine until we want to bind to resources in XAML. For some reason, our resources class must be `public` for this to work (see [the guide](https://docs.avaloniaui.net/docs/guides/implementation-guides/localizing)). If we don't, the error message claims the member just *doesn't exist*:

> Error AVLN2000 Avalonia: Unable to resolve "Resources.CancelText" as static field, property, constant or enum value Line 32, position 17.

Which is bad (it should say it exists, but has wrong accessibility), but that's a different discussion. *Why* can't Avalonia bind to internal members? Especially because the lowered XAML is just `target2.Text = MyNamespace.Resources.CancelText`. Besides, IL can ignore accessibility modifiers, so they shouldn't matter. Why must I expose my internal `Resources` type to DLL users just so Avalonia can work?

**Comment by maxkatz6** [comment]
`x:Static` is not related to binding, but it still should work with internal members **in the same assembly**. Otherwise, it's a bug.

**Reply by colejohnson66** [reply]
Thanks for clarifying. I've opened a bug report: #16929.
